PLANET-7535 Credit is removed when caption is edited

Osong-Michael opened the pull request:
**Ref: https://jira.greenpeace.org/browse/PLANET-7535**

**Testing:** 

On local or assigned test-instance, when image captions is edited, credits should still be visible.

This handles two scenarios as implemented.
 - Edit an image in the Media Library. Then add it in the post. The caption should be the new one you just edited.
 - Edit an image directly in the editor. Then the caption displayed should be the one you just edited in the post regardless of what's stored in the Media Library.

<!-- Ref: Please add a url to the ticket this change is addressing.

---

Please provide a brief summary of the change introduced to make review process easier.

Ideally this should also be part of the commit summary.
-->

planet-4 commented:
### Test instance is ready :rocket:

:new_moon: [janus](https://www-dev.greenpeace.org/test-janus) | [admin](https://www-dev.greenpeace.org/test-janus/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-janus/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-janus) | [composer-local.json](https://github.com/greenpeace/planet4-test-janus/blob/main/composer-local.json)

:watch: 2024.06.11 10:31:14

mleray commented:
I tested it on [this page](https://www-dev.greenpeace.org/test-janus/coucou/), it works as expected if I update the caption in the Library 👍 But if I edit it directly in the editor under the image, the old value still shows in the frontend... The credits still show though

GP-Dan-Tovbein commented:
I've tested on test instance but not sure if it works as expected. Maybe I'm doing something wrong?


https://github.com/greenpeace/planet4-master-theme/assets/77975803/8cd3a169-ab01-495a-9df9-325703bb260e

Osong-Michael commented:
> I've tested on test instance but not sure if it works as expected. Maybe I'm doing something wrong?
> 
>  Screen.Recording.2024-06-12.at.06.26.55.mov

I just updated the code, you can test again once the pipeline runs. I also updated the PR comment to reflect the changes

mleray reviewed (approved):
Tested on the instance, seems to work as expected 👍
